modin/numpy/arr.py

```diff
@@ -408,71 +411,88 @@ def max(
         if self._ndim == 1:
             if axis == 1:
                 raise numpy.AxisError(1, 1)
-            result = self._query_compiler.max(axis=0)
+            result = self._query_compiler
+            if isinstance(where, array):
+                result = where.where(self, initial)._query_compiler
+            result = result.max(axis=0)
             if keepdims:
-                if initial is not None and result.lt(initial):
+                if initial is not None and result.lt(initial).any():
                     result = pd.Series([initial])._query_compiler
-                if initial is not None:
-                    if out is not None:
-                        out._query_compiler = (
-                            numpy.ones_like(out) * initial
-                        )._query_compiler
-                    else:
-                        out = array([initial]).astype(self.dtype)
+                if initial is not None and out is not None:
+                    out._query_compiler = (
+                        numpy.ones_like(out) * initial
+                    )._query_compiler
                 if out is not None and out.shape != (1,):
                     raise ValueError(
                         f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
                     )
-                return fix_dtypes_and_determine_return(result, 1, dtype, out, where)
+                return (
+                    fix_dtypes_and_determine_return(
+                        result, 1, dtype, out, (where is not False)
+                    )
+                    if where is not False or out is not None
+                    else array([initial])
+                )
```

Comment:
```suggestion
                if where is not False or out is not None:
                    return fix_dtypes_and_determine_return(result, 1, dtype, out, where is not False)
                return array([initial])
```
I wonder if this reads better...